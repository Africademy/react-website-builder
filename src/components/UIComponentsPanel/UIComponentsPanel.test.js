import React from 'react';
import UIComponentPanel from '.';
import TestRenderer from 'react-test-renderer';

const vals = {
    section1 : {
        name : "Comps",
        values : [
            "A",
            "B",
            "C",
            "A",
            "B",
            "C"
        ]
    },
    section2 : {
        name : "Others",
        values : [
            "A",
            "B"
        ]
    }
}

test("show/hide components when section is clicked", ()=> {
    const comp = TestRenderer.create(
        <UIComponentPanel sections={vals} />,
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();

//     tree.props.handleExpansion();
//     tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
});