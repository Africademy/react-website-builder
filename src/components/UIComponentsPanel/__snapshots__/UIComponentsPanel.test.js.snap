// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show/hide components when section is clicked 1`] = `
Array [
  <div
    className="section"
  >
    <button
      aria-controls="Comps"
      aria-expanded={true}
      className="section-btn btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Comps
    </button>
    <div
      aria-expanded={null}
      className="collapse show container"
      id="Comps"
    >
      <div
        className="justify-content-center row"
      >
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          A
        </button>
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          B
        </button>
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          C
        </button>
      </div>
      <div
        className="justify-content-center row"
      >
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          A
        </button>
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          B
        </button>
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          C
        </button>
      </div>
    </div>
  </div>,
  <div
    className="section"
  >
    <button
      aria-controls="Others"
      aria-expanded={true}
      className="section-btn btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Others
    </button>
    <div
      aria-expanded={null}
      className="collapse show container"
      id="Others"
    >
      <div
        className="justify-content-center row"
      >
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          A
        </button>
        <button
          className="m-2 btn btn-outline-success"
          disabled={false}
          type="button"
        >
          B
        </button>
      </div>
    </div>
  </div>,
]
`;
